File: c/cz/mg/vulkan/PFN_vkAllocateMemory.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkAllocateMemory_call(JNIEnv* env, jclass jc, jlong address, jlong device, jlong pAllocateInfo, jlong pAllocator, jlong pMemory, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkAllocateMemory f = (PFN_vkAllocateMemory)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        ((VkMemoryAllocateInfo*)jniLongToPointer(pAllocateInfo)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator)),
        ((VkDeviceMemory*)jniLongToPointer(pMemory))
    );
}

File: cz/mg/vulkan/PFNvkAllocateMemory.java
package cz.mg.vulkan;

public class PFNvkAllocateMemory extends VkFunctionPointer {
    public PFNvkAllocateMemory() {
    }

    public PFNvkAllocateMemory(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkAllocateMemory(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkAllocateMemory(long value) {
        setValue(value);
    }

    public PFNvkAllocateMemory(VkInstance instance) {
        super(instance, new VkString("vkAllocateMemory"));
    }

    public void call(VkDevice device, VkMemoryAllocateInfo pAllocateInfo, VkAllocationCallbacks pAllocator, VkDeviceMemory pMemory, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.getNullAddress(), pAllocateInfo != null ? pAllocateInfo.getVkAddress() : VkPointer.NULL, pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL, pMemory != null ? pMemory.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.getNullAddress());
    }

    private static native void call(long vkaddress, long device, long pAllocateInfo, long pAllocator, long pMemory, long rval);
}
